### PR TITLE
OAuth authorization implemented with Github and Google

### PR DESCRIPTION
**Pull request by fructusmortus**
authMiddleware fixed (with de serialized user from cookies), some documentation added

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,6 +23,8 @@ https://lixpi.docs.apiary.io
 
 Github authentication steps
 ```
-Run in browser http://localhost:3005/connect/github
-Log in with your credentials
-As result forward to http://localhost:3005/testauth (callback url), which demonstrates public user details
\ No newline at end of file
+Create and configure an app in Developer Settings -> OAuth Apps.
```
Comment (shelby-carter) [inline README.md:26]: One step is missing, first you have to navigate to `Developer settings` on the left nav menu in user settings and then from there you'll be able to navigate to `OAuth Apps`

Comment (fructusmortus) [reply README.md]: Done

File: README.md [hunk 2]
```diff
@@ -23,6 +23,8 @@ https://lixpi.docs.apiary.io
...
+Homepage URL is our API address http://localhost:3005/.
+Authorization callback URL (as we use the Grant npm) will be http://localhost:3005/connect/github/callback (it's a mandatory pattern).
```
Comment (shelby-carter) [inline README.md:29]: Missing step about generating a secret key on the GitHub OAuth app settings page and copying and pasting both `Client ID` and `Secret Key` to `config/<env-type>.env` file

File: README.md [hunk 3]
```diff
@@ -23,6 +23,8 @@ https://lixpi.docs.apiary.io
...
\ No newline at end of file
+Create and configure an app in Developer Settings -> OAuth Apps.
+Homepage URL is our API address http://localhost:3005/.
+Authorization callback URL (as we use the Grant npm) will be http://localhost:3005/connect/github/callback (it's a mandatory pattern).  
+In server.js we imported the Grant, connected it with the Express and the Session. Then passed it with a github config file, where were set an origin host URL, key, secret, callback URL ('/login', yes the're different with the callback URL in App's settings) and a response type.
+When we got a request on the login route it executes auth middleware, where we can get all of the public user's data such as email, name, avatar etc. Then it creates User (if not exists) and Session records in a database. After that it creates an accessToken and a refreshToken for a current Session as JWTs. To create JWTs we use the Jsonwebtoken npm, with a pair of private and public keys which generated with ssh-keygen by algoritm RS256 (<script src="https://gist.github.com/ygotthilf/baa58da5c3dd1f69fae9.js"></script>)
```
Comment (shelby-carter) [inline README.md:31]: Could you please include actual commands that have to be executed to generate the key. Also it's unclear where to put the key and in which format.

File: README.md (hunk 2 above)
Comment (fructusmortus) [reply README.md]: Done

File: README.md (hunk 3 above)
Comment (fructusmortus) [reply README.md]: Done

**Pull request merged**